Lib/test/datetimetester.py

```diff
@@ -656,11 +656,31 @@ def test_str(self):
     def test_repr(self):
         name = 'datetime.' + self.theclass.__name__
         self.assertEqual(repr(self.theclass(1)),
-                         "%s(1)" % name)
+                         "%s(days=1)" % name)
         self.assertEqual(repr(self.theclass(10, 2)),
-                         "%s(10, 2)" % name)
+                         "%s(days=10, seconds=2)" % name)
         self.assertEqual(repr(self.theclass(-10, 2, 400000)),
-                         "%s(-10, 2, 400000)" % name)
+                         "%s(days=-10, seconds=2, microseconds=400000)" % name)
+        self.assertEqual(repr(self.theclass(1, -1)),
+                         "%s(seconds=86399)" % name)
+        self.assertEqual(repr(self.theclass(0, 1, -1)),
+                         "%s(microseconds=999999)" % name)
+        self.assertEqual(repr(self.theclass(0, 86400)),
+                         "%s(days=1)" % name)
+        self.assertEqual(repr(self.theclass(0, 2 * 86400)),
+                         "%s(days=2)" % name)
+        self.assertEqual(repr(self.theclass(days=1, seconds=0)),
+                         "%s(days=1)" % name)
+        self.assertEqual(repr(self.theclass(seconds=60)),
+                         "%s(seconds=60)" % name)
+        self.assertEqual(repr(self.theclass()),
+                         "%s(seconds=0)" % name)
+        self.assertEqual(repr(self.theclass(microseconds=100)),
+                         "%s(microseconds=100)" % name)
+        self.assertEqual(repr(self.theclass(days=1, microseconds=100)),
+                         "%s(days=1, microseconds=100)" % name)
+        self.assertEqual(repr(self.theclass(seconds=1, microseconds=100)),
+                         "%s(seconds=1, microseconds=100)" % name)
```

Comment:
I don't think that it's worth it to add so many test cases. Just keep 3 examples, 1 negative value, the timedelta(0), and you're done no?